.ci/kata-simplify-log.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# Copyright (c) 2018 Intel Corporation
+#
+# SPDX-License-Identifier: Apache-2.0
+
+readonly script_name=${0##*/}
+
+die()
+{
+    local -r msg="$*"
+    echo >&2 "ERROR: $msg"
+    exit 1
+}
+
+usage()
+{
+    cat <<EOT
+Description: Simplify the specified logfile by replacing common fields with
+fixed strings to make diff(1)-ing easier.
+
+Usage: $script_name <log-file>
+       $script_name [-h|--help|help]
+
+Options:
+
+  -h     : Show this help.
+  --help :
+  help   :
+
+Limitations:
+
+- This script uses simple heuristics and might break at any time.
+
+EOT
+}
+
+simplify_log()
+{
+    local -r file="$1"
+
+    sed -r \
+        -e 's/[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{,9}[+-][0-9]{2}:[0-9]{2}/TIMESTAMP/gI' \
+        -e 's/[0-9]{4}-[0-9]{2}-[0-9]{2}T[0-9]{2}:[0-9]{2}:[0-9]{2}.[0-9]{,9}Z/TIMESTAMP/gI' \
+        -e 's/"[0-9][0-9]* kB/MEMORY-SIZE/gI' \
+        -e 's/[[:xdigit:]]{64}-[[:xdigit:]]{16}-/MOUNT-HASH/gI' \
+        -e 's/[[:xdigit:]]{64}/HASH/gI' \
+        -e 's/duration=[^ ][^ ]* /duration=DURATION/gI' \
+        -e 's/0x[[:xdigit:]]{,10}/0xADDR/gI' \
+        -e 's/[[:xdigit:]]{8}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{4}-[[:xdigit:]]{12}/UUID/gI' \
+        -e 's/[[:xdigit:]]{2}:[[:xdigit:]]{2}:[[:xdigit:]]{2}:[[:xdigit:]]{2}:[[:xdigit:]]{2}:[[:xdigit:]]{2}/MAC-ADDRESS/gI' \
+        -e 's/[[:xdigit:]]{40}/COMMIT/gI' \
+        -e 's![0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}/[0-9]{1,3}!IP-ADDRESS-AND-MASK!gI' \
+        -e 's![0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}.[0-9]{1,3}!IP-ADDRESS!gI' \
+        -e 's/pid=[0-9][0-9]*/pid=PID/gI' \
+        -e 's!/proc/[0-9][0-9]*/!/proc/PID/!gI' \
+        -e 's/returned -*[0-9][0-9]* after [0-9][0-9]* usecs/returned VALUE after VALUE usecs/g' \
+        -e 's/[0-9][0-9]*/NUMBER/gI' \
+        "$file"
+}
+
+[ $# -eq 1 ] || die "need log file"
```

Comment:
Maybe print the help here instead - I didn't realise the script had help until I read the source ;-)